tests/publiccloud/ssh_interactive_start.pm

```diff
@@ -16,17 +16,39 @@ use Mojo::Base 'publiccloud::ssh_interactive_init';
 use publiccloud::ssh_interactive;
 use testapi;
 use utils;
+use publiccloud::utils "select_host_console";
 
 sub run {
     my ($self, $args) = @_;
-    select_console 'tunnel-console';
+
+    # This ensure that we have used the setup console, even if no module was run before.
```

Comment:
Minor suggestion: What do you think to move lines 24-38 into it's own routine for better understanding? I'm thinking of just creating a `sub establish_tunnel` or something similar.

Reply:
yes I think some further refactoring would be nice.

But we need to decide which direction. 
we have
* `lib:publiccloud:ssh_interactive_init` , which is a base class for:
  *  `tests/publiccloud/patch_and_reboot.pm`
  *  `tests/publiccloud/transfer_repos.pm`
  * `tests/publiccloud/register_system.pm`
  *  `tests/publiccloud/ssh_interactive_start.pm`
* `lib::publiccloud:ssh_interactive` which contain utility functions (each function is only called once!)
* and the connected tests:
  *  `tests/publiccloud/ssh_interactive_init.pm` (connect to PC instance and do preparations)
  *  `tests/publiccloud/ssh_interactive_start.pm` (final step to use `TUNNELED` mode of **all** consoles)
  *  `tests/publiccloud/ssh_interactive_end.pm` (cleanup and revert `TUNNELED` after all modules)

Things I like to cover:
* Don't use same name in test-module and lib
* Use a common baseclass for `ssh_interacive_(init|start|end).pm`
* Decide, if we wanna have the logic in some lib-function or directly visible in the test-module

Reply:
And actually it would be nice to have a setup-console selection acrostic approach for `TUNNELED` behavior.
Something like
 * remember which console was already activated while `_TUNNELED_SETUP_IN_PROGRESS` is set
 * run over all consoles or do on select console the `ssh -t sut` when `_SSH_TUNNELS_INITIALIZED` is set

But I didn't found a good solution, thats why I would like to postpone it with this PR

@okurz  what do you think?

Reply:
I guess it's merely the decision between keeping the "remote ssh tunnel" approach more explicit and hence different or streamline it as part of what we consider the normal console based approach for all cases. I am not sure who you would want it to look like. But feel welcome to go ahead :)